tests/vmi_multus_test.go

```diff
@@ -375,32 +375,48 @@ var _ = Describe("[Serial]Multus", func() {
 		})
 
 		Context("VirtualMachineInstance with Linux bridge CNI plugin interface and custom MAC address.", func() {
-			interfaces := []v1.Interface{linuxBridgeInterface}
-			networks := []v1.Network{linuxBridgeNetwork}
-			linuxBridgeIfIdx := 0
 			customMacAddress := "50:00:00:00:90:0d"
 
 			It("[test_id:676]should configure valid custom MAC address on Linux bridge CNI interface.", func() {
 				By("Creating a VM with Linux bridge CNI network interface and default MAC address.")
-				vmiTwo := createVMIOnNode(interfaces, networks)
-				tests.WaitUntilVMIReady(vmiTwo, console.LoginToAlpine)
+				vmiTwo := libvmi.NewFedora(
+					libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+					libvmi.WithNetwork(v1.DefaultPodNetwork()),
+					libvmi.WithInterface(linuxBridgeInterface),
+					libvmi.WithNetwork(&linuxBridgeNetwork),
+					libvmi.WithCloudInitNoCloudUserData(tests.GetGuestAgentUserData(), false),
+					libvmi.WithCloudInitNoCloudNetworkData(cloudInitNetworkDataWithStaticIPsByDevice("eth1", "10.1.1.2/24"), false))
+				vmiTwo = tests.StartVmOnNode(vmiTwo, nodes.Items[0].Name)
 
 				By("Creating another VM with custom MAC address on its Linux bridge CNI interface.")
-				interfaces[linuxBridgeIfIdx].MacAddress = customMacAddress
-				vmiOne := createVMIOnNode(interfaces, networks)
-				tests.WaitUntilVMIReady(vmiOne, console.LoginToAlpine)
-
-				By("Configuring static IP address to the Linux bridge interface.")
-				Expect(configInterface(vmiOne, "eth0", "10.1.1.1/24")).To(Succeed())
-				Expect(configInterface(vmiTwo, "eth0", "10.1.1.2/24")).To(Succeed())
+				linuxBridgeInterfaceWithCustomMac := linuxBridgeInterface
+				linuxBridgeInterfaceWithCustomMac.MacAddress = customMacAddress
+				vmiOne := libvmi.NewFedora(
+					libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+					libvmi.WithNetwork(v1.DefaultPodNetwork()),
+					libvmi.WithInterface(linuxBridgeInterfaceWithCustomMac),
+					libvmi.WithNetwork(&linuxBridgeNetwork),
+					libvmi.WithCloudInitNoCloudUserData(tests.GetGuestAgentUserData(), false),
+					libvmi.WithCloudInitNoCloudNetworkData(cloudInitNetworkDataWithStaticIPsByMac(customMacAddress, "10.1.1.1/24"), false))
+				vmiOne = tests.StartVmOnNode(vmiOne, nodes.Items[0].Name)
+
+				tests.WaitUntilVMIReady(vmiOne, console.LoginToFedora)
+				tests.WaitAgentConnected(virtClient, vmiOne)
+				tests.WaitUntilVMIReady(vmiTwo, console.LoginToFedora)
+				tests.WaitAgentConnected(virtClient, vmiTwo)
+
+				var interfaceName string
+				Eventually(func() error {
+					var err error
+					vmiOne, err = virtClient.VirtualMachineInstance(vmiOne.Namespace).Get(vmiOne.Name, &metav1.GetOptions{})
+					Expect(err).NotTo(HaveOccurred())
+					interfaceName, err = getInterfaceNameByMAC(vmiOne, customMacAddress)
+					return err
+				}, 140*time.Second, 5*time.Second).Should(Succeed())
```

Comment:
I do not understand this one.
- Why do we need it at all?
- Whey do we need a retry here?

Reply:
> Why do we need it at all?
The order of interfaces at vmiOne.Status.Interfaces can change so we need to search in them to get the proper interface name, @alonSadan put in place some nice helpers to do so.

> Whey do we need a retry here?
Interface name depends on Guest Agent and we know it takes some time for it to kick in.

Reply:
I think that since i mixed here, and it disregard GA as you said, 
both of the issues won't occur ?

Reply:
> The order of interfaces at vmiOne.Status.Interfaces can change so we need to search in them to get the proper interface name, @alonSadan put in place some nice helpers to do so.

We enforce the name-mac pair with cloud-init, so looking for the iface name to match the mac is not needed.

> 
> > Whey do we need a retry here?
> 
> Interface name depends on Guest Agent and we know it takes some time for it to kick in.

Interesting, I am not sure this is true or not, but the previous answer may make it irrelevant anyway.
(i.e what happens if GA is not there, we do not report iface name?)

Reply:
@oshoval going to put back my original code with the twist of using the network name as the interface name.